enterprise/server/backends/distributed/distributed.go

```diff
@@ -390,24 +396,29 @@ func dedupe(in []string) []string {
 // peers are returned in random order.
 func (c *Cache) readPeers(d *repb.Digest) *peerset.PeerSet {
 	peers := c.consistentHash.GetAllReplicas(d.GetHash())
-	primaryPeers := peers[:c.config.ReplicationFactor]
-	secondaryPeers := peers[c.config.ReplicationFactor:]
+	var primaryPeers, secondaryPeers []string
+	if len(peers) >= c.config.ReplicationFactor {
```

Comment:
would be nice to add a tiny comment here explaining this logic

// To prevent a panic if replication is misconfigured to be higher than peer count.